src/dataframe/dataframe.jl

```diff
@@ -1096,14 +1115,18 @@ function Base.append!(df1::DataFrame, df2::AbstractDataFrame)
 
     _names(df1) == _names(df2) || error("Column names do not match")
     nrows, ncols = size(df1)
+    targetrows = nrows + nrow(df2)
     try
         for j in 1:ncols
             append!(df1[!, j], df2[!, j])
         end
+        for col in _columns(df1)
+            @assert length(col) == targetrows
```

Comment:
Same here.

Reply:
Like above I personally would prefer to see `@assertion` error here to know that something really bad happened (opposed to simply passing a wrong argument).